packages/core/src/lib/connection_manager.ts

```diff
@@ -340,7 +340,11 @@ export class ConnectionManager
       void (async () => {
         const peerId = evt.detail;
 
-        this.keepAliveManager.start(peerId, this.libp2p.services.ping);
+        this.keepAliveManager.start(
+          peerId,
```

Comment:
nit: we can simplify and pass ‘libp2p’ directly

Reply:
SOLID is not applied here because it is more about how entity/object should be designed and not what it should be provided with. You might need thousand dependencies to do single thing and it is up to devs how easier/cleaner they should be provided. 

My rule of thumb for dependencies - it's easier to grow the number of entities than methods. 

And again, it is a nit, so feel free to keep the way you think reasonable.

Reply:
No, it's exactly about necessary dependencies, and the necessary dependency here is `libp2p` and not its methods.